src/ws/ws_manager.rs

```diff
@@ -65,13 +67,22 @@ pub(crate) struct SubscriptionSendData<'a> {
     subscription: &'a serde_json::Value,
 }
 
+#[derive(Serialize)]
+pub(crate) struct Ping {
+    method: &'static str,
+}
+
 impl WsManager {
+    const SEND_PING_INTERVAL: u64 = 50;
+
     pub(crate) async fn new(url: String) -> Result<WsManager> {
         let (ws_stream, _) = connect_async(url.clone())
             .await
             .map_err(|e| Error::Websocket(e.to_string()))?;
 
         let (writer, mut reader) = ws_stream.split();
+        let writer = Arc::new(Mutex::new(writer));
+        let writer_copy = Arc::clone(&writer);
```

Comment:
nit: wrap the `ping_fut` block with brackets and then you can write `let writer = Arc::clone(&writer)` thereby avoiding having to write a variable suffixed with `_copy`

Reply:
Done.